src/com/photoncat/aiproj2/heuristic/Heuristic2.java

```diff
@@ -0,0 +1,902 @@
+package com.photoncat.aiproj2.heuristic;
+
+import com.photoncat.aiproj2.interfaces.Board;
+import com.photoncat.aiproj2.interfaces.Board.PieceType;
+import com.photoncat.aiproj2.interfaces.Heuristics;
+
+public class Heuristic2 implements Heuristics {
+    @Override
+    public int heuristic(Board board){
+        int n=board.getSize();
+        int m=board.getM();
+
+        if (n==3&&m==3){
+            return ThreeByThree(board);
+        }
+        if (m==5){
+            return NByFive(board);
+        }
+        if (m==4){
+            return NByFour(board);
+        }
+        else{
+            return NbyM(board);
+        }
+//        if ()
+//        return 0;
+    }
+
+    private int NbyM(Board board) {
+        if (checkWin(board,PieceType.CROSS)){
```

Comment:
The board caches who is winning by board.wins(). Please utilize that function instead of implementing one your own.